regex-automata/tests/nfa/thompson/backtrack/suite.rs

```diff
@@ -108,8 +108,18 @@ fn compiler(
         if !configure_backtrack_builder(test, &mut builder) {
             return Ok(CompiledRegex::skip());
         }
-        let re = builder.build_many(&regexes)?;
-        // TODO: remove once look-around is supported.
+        let re = match builder.build_many(&regexes) {
+            Ok(re) => re,
+            // Due to errors being opaque, we need to check the error message to skip tests with look-arounds
+            Err(err) => {
+                if test.compiles() && err.to_string().contains("look-around") {
+                    return Ok(CompiledRegex::skip());
+                }
+
+                return Err(err.into());
+            }
+        };
+        // The backtracker doesn't support lookarounds, so skip if there are any.
         if re.get_nfa().lookaround_count() > 0 {
             return Ok(CompiledRegex::skip());
         }
```

Comment:
Is the match block really necessary? Couldn't we move the if block above the `build_many` call and be done with it?

Reply:
Unfortunately it uses the `re` which is defined above. Before that, we don't have any NFA. However, now the if condition is not needed at all I think.

Reply:
ah yeah, correct